encoder.go

```diff
@@ -75,6 +70,11 @@ func (e *Encoder) encode(v reflect.Value, dst map[string][]string) error {
 			continue
 		}
 
+		if v.Field(i).Type().Kind() == reflect.Struct {
```

Comment:
I'm OK with this approach, but can you add tests for this? (that the order is respected)

@kisielk for review as well.